### PR TITLE
Fix listVPCs unmarshall error when a VPC has network tiers attached

### PR DESCRIPTION
**Pull request by tetra12**
This is the fix for #25

### CODE REVIEW

**Comment by tetra12** [comment]
This is actually a quick and dirty fix, since I'm fixing schema in a generated file :roll_eyes: . I'll be looking :eyes: into this how to properly change the generator code to get a valid generated schema

**Comment by tetra12** [comment]
@svanharmelen Hello :clap:
Could you pls review my PR? It seems like devs are not active here. Probably due to the knowledge gap in this project

**Comment by tetra12** [comment]
@svanharmelen It looks like the scheme has diverged in 4.16. Definitely need to update the generator code

**Comment by svanharmelen** [comment]
Hi @tetra12, thanks for reaching out but I'm not actively involved in this project anymore (hence the reason I transferred it). So I guess you need to be patient and wait for someone from the team to pick this up and respond... Sorry!

**Comment by tetra12** [comment]
@rohityadavcloud Hi Rohid :clap: . Is there anyone in charge of maintaining this project? I'd like to contribute but see noone to be engaged after the the project transfer.

File: cloudstack/VPCService.go [hunk 1]
```diff
@@ -2855,7 +2855,7 @@ type VPC struct {
 	JobID                string               `json:"jobid"`
 	Jobstatus            int                  `json:"jobstatus"`
 	Name                 string               `json:"name"`
-	Network              []string             `json:"network"`
+	Network              []Network            `json:"network"`
```
Comment (Pearl1594) [inline cloudstack/VPCService.go:2858]: @tetra12 - the change LGTM, but I believe this change would also probably be required at `CreateVPCResponse` (https://github.com/apache/cloudstack-go/blob/main/cloudstack/VPCService.go#L721) and `UpdateVPCResponse` (https://github.com/apache/cloudstack-go/blob/main/cloudstack/VPCService.go#L3172). You could probably add a check in the function named `mapType` (https://github.com/apache/cloudstack-go/blob/main/generate/generate.go#L1849) - to handle this change via the code generator. Thanks.

Comment (tetra12) [reply cloudstack/VPCService.go:2858]: Hi @Pearl1594 
Thanks for your hint. I updated the generator code as per your advice. Pls check it up again

**Review by Pearl1594 — APPROVED** [review]
LGTM. Thanks @tetra12.

**Comment by tetra12** [comment]
Anyone alive here ? Is there anyone to merge a PR ?

**Review by harikrishna-patnala — APPROVED** [review]
Code changes LGTM, did not test it.

**Comment by kohrar** [comment]
Thanks for making this PR. I can confirm that it fixes an issue with the Terraform CloudStack provider when used against CloudStack 4.16 where the provider crashes when trying to parse the VPC output.

**Comment by rohityadavcloud** [comment]
@Pearl1594 is this good to merge, or we need to do any further testing. cc @davidjumani - if this affects any of your work on cks/capc?

**Comment by Pearl1594** [comment]
Yes @rohityadavcloud - it's good to go.

**Comment by rohityadavcloud** [comment]
@davidjumani good to merge as well?

File: generate/generate.go [hunk 2]
```diff
@@ -1849,6 +1849,8 @@ func mapType(aName string, pName string, pType string) string {
 	case "list":
 		if pName == "downloaddetails" || pName == "owner" {
 			return "[]map[string]string"
+		} else if pName == "network" {
+			return "[]Network"
```
Comment (davidjumani) [inline generate/generate.go:1853]:
```suggestion
			return "[]*Network"
```

**Pull request merged**